src/librbd/mirror/snapshot/RemoveImageStateRequest.cc

```diff
@@ -0,0 +1,127 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "librbd/mirror/snapshot/RemoveImageStateRequest.h"
+#include "common/dout.h"
+#include "common/errno.h"
+#include "librbd/ImageCtx.h"
+#include "librbd/Utils.h"
+#include "librbd/mirror/snapshot/Types.h"
+#include "librbd/mirror/snapshot/Utils.h"
+
+#define dout_subsys ceph_subsys_rbd
+
+#undef dout_prefix
+#define dout_prefix *_dout << "librbd::mirror::snapshot::RemoveImageStateRequest: " \
+                           << this << " " << __func__ << ": "
+
+namespace librbd {
+namespace mirror {
+namespace snapshot {
+
+using librbd::util::create_rados_callback;
+
+template <typename I>
+void RemoveImageStateRequest<I>::send() {
+  get_object_count();
+}
+
+
+template <typename I>
+void RemoveImageStateRequest<I>::get_object_count() {
+  CephContext *cct = m_image_ctx->cct;
+
+  auto oid = util::image_state_object_name(m_image_ctx, m_snap_id, 0);
+  ldout(cct, 20) << oid << dendl;
+
+  librados::ObjectReadOperation op;
+  op.read(0, ImageStateHeader::length(), &m_bl, nullptr);
```

Comment:
You can just read the first 1 + 1 + 4 + 4 bytes. The decode below would fail if this is v1 code but v2 is written w/ a larger than read struct len since decode would attempt to skip until the end (and hit an end-of-buffer exception).